src/pages/home/report/ContextMenu/ReportActionContextMenu.ts

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import {GestureResponderEvent, Text as RNText} from 'react-native';
+import {OnyxEntry} from 'react-native-onyx';
+import {ValueOf} from 'type-fest';
+import type {ReportAction} from '@src/types/onyx';
+// eslint-disable-next-line import/no-cycle
+import {CONTEXT_MENU_TYPES} from './ContextMenuActions';
+
+type OnHideCallback = () => void;
+
+type OnConfirm = () => void;
+
+type OnCancel = () => void;
+
+type ContextMenuType = ValueOf<typeof CONTEXT_MENU_TYPES>;
+
+type ShowContextMenu = (
+    type: ContextMenuType,
+    event: GestureResponderEvent | MouseEvent,
+    selection: string,
+    contextMenuAnchor: RNText | null,
+    reportID?: string,
+    reportActionID?: string,
+    originalReportID?: string,
+    draftMessage?: string,
+    onShow?: () => void,
+    onHide?: () => void,
+    isArchivedRoom?: boolean,
+    isChronosReport?: boolean,
+    isPinnedChat?: boolean,
+    isUnreadChat?: boolean,
+) => void;
```

Comment:
Can we improve arguments here and use object?

```suggestion
type ShowContextMenu = ({
    type: ContextMenuType,
    event: GestureResponderEvent | MouseEvent,
    selection: string,
    contextMenuAnchor: RNText | null,
    reportID?: string,
    reportActionID?: string,
    originalReportID?: string,
    draftMessage?: string,
    onShow?: () => void,
    onHide?: () => void,
    isArchivedRoom?: boolean,
    isChronosReport?: boolean,
    isPinnedChat?: boolean,
    isUnreadChat?: boolean,
}) => void;
```

Reply:
@rezkiy37 Since this update will lead to updates in several files and will need to be additionally described and tested, it's better to skip it in this PR. @fabioh8010 What do you think?